### PR TITLE
Extending Abstract classes for ServiceMapProcessor, BlockingBuffer, a…

### PR DESCRIPTION
**Pull request by AustinTag**
…nd ESSink

*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/Data-Prepper/issues/201

*Description of changes:*
- Extending AbstractPrepper for ServiceMapStatefulProcessor
  - Still todo: add custom metrics
- Extending AbstractSink for ElasticsearchSink
  - Still todo: custom metrics
- Extending AbstractBuffer for BlockingBuffer
- Adding new constructor in AbstractBuffer to support BlockingBuffer constructors
- Adding new static method to PluginMetrics to create an instance without a PluginSetting
- Adding pipeline name to PluginSetting in ES integration test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by chenqi0805 — APPROVED** [review]
LGTM

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java [hunk 1]
```diff
@@ -46,5 +46,5 @@ public AbstractPrepper(final PluginSetting pluginSetting) {
      * @param records Input records
      * @return Processed records
      */
```
Comment (yadavcbala) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java:48]: I asked this question to Technical Writer but haven't got any response - Are we changing the name of the component from `Processor` to `Prepper` ? [just for my information, I would like to leave it as Processor but ok with Prepper]

File: data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import com.amazon.dataprepper.model.PluginType;
 import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
 import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.processor.AbstractPrepper;
 import com.amazon.dataprepper.model.processor.Processor;
```
Comment (yadavcbala) [inline data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java:7]: remove unused import

File: data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ElasticsearchSink.java [hunk 3]
```diff
@@ -4,6 +4,7 @@
 import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
 import com.amazon.dataprepper.model.configuration.PluginSetting;
 import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.model.sink.AbstractSink;
 import com.amazon.dataprepper.model.sink.Sink;
```
Comment (yadavcbala) [inline data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ElasticsearchSink.java:8]: unused import

File: data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java [hunk 4]
```diff
@@ -24,7 +25,7 @@
 import java.util.stream.Stream;
 
 @DataPrepperPlugin(name = "service_map_stateful", type = PluginType.PROCESSOR)
-public class ServiceMapStatefulProcessor implements Processor<Record<ExportTraceServiceRequest>, Record<String>> {
+public class ServiceMapStatefulProcessor extends AbstractPrepper<Record<ExportTraceServiceRequest>, Record<String>> {
```
Comment (yadavcbala) [inline data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java:28]: Do you want to update the name to `ServiceMapStatefulPrepper` - does not like the sounding but just asking if you intent to ?

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java (hunk 1 above)
Comment (yadavcbala) [reply data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java:48]: I think we should keep it as `AbstractProcessor` - but I am fine if everyone wants to update it to `AbstractPrepper`

**Review by yadavcbala — COMMENTED** [review]
No major blocking comments - but a naming confirmation, will approve after confirming

**Review by yadavcbala — APPROVED** [review]
Non-Blocking nit comments : Name `AbstractPrepper` is decided per @kowshikn

File: data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java (hunk 2 above)
Comment (AustinTag) [reply data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java:7]: Good catch, will do

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java (hunk 1 above)
Comment (AustinTag) [reply data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java:48]: I am fine either way, lets just all align on one name

File: data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java (hunk 4 above)
Comment (AustinTag) [reply data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java:28]: Sure, I'll do that

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java (hunk 1 above)
Comment (kowshikn) [reply data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractPrepper.java:48]: https://github.com/opendistro-for-elasticsearch/Data-Prepper/issues/209 :)

**Pull request merged**